src/lib/web_channel_event_recorder.js

```diff
@@ -0,0 +1,85 @@
+import app from 'ampersand-app';
+
+class WebChannelEventRecorder {
+  constructor () {
+    this.namespace = 'wcer-';
+    this.maxEvents = 100;
+  }
+
+  record () {
+    window.addEventListener('WebChannelMessageToContent', this._messageReceived.bind(this));
+  }
+
+  replay (eventName, max = 1, offset = 0) {
+    let details = this._getItem(eventName);
+
+    if (!details) {
+      return;
+    }
+
+    let eventsToReplay = max > 0 ? details.slice(offset, offset + max) : details;
+
+    eventsToReplay.forEach((detail) => { this._triggerMessageToContent(detail); });
+  }
+
+  replayAll (eventName) {
+    this.replay(eventName, -1);
+  }
+
+
+  enable () {
+    return localStorage.setItem('WebChannelEventRecorder.enabled', 'true');
```

Comment:
If this was for more than debugging I'd agree, but probably okay for it to crash hard if there's no `localStorage` in this case.